docs/userGuide/syntax/code.mbdf

```diff
@@ -58,42 +58,100 @@ function add(a, b) {
 
 
 ##### Line highlighting
-To highlight lines, add the attribute `highlight-lines` as shown below.
 
-You can specify highlighting in many different ways, depending on how you want it to be. There are two main variants:
+You can add the `highlight-lines` attribute to add highlighting to your code block. Refer to the example below for
+all possible ways of highlighting a code block.
 
-**Text-only highlighting**
+The value of `highlight-lines` is composed of *highlight rules*, separated by commas.
+These rules dictate where and how MarkBind should highlight your code block.
 
-To highlight only the text portion of the line, you can just use the line numbers as is.
+You can specify the rules in many different ways, depending on how you want it to be. There are three main variants:
+full text, substring, bounded (character-wise or word-wise), or full line highlighting.
```

Comment:
seems a little lengthy; Could we compress all this into a table? 🤔

Reply:
Yeah I agree :/ Felt like a lot to explain from the various types, syntax formats, and its limitations. But not sure how far I can condense it to a table, looks like some parts may need to just have a brief sentence or two without making the table feel like a wall of text. I'll try to do it soon.

Reply:
Looks really neat now 👍 thanks!